src/Makefile-env.am

```diff
@@ -118,28 +118,30 @@ AM_COMMON_CFLAGS = \
 	-fno-strict-aliasing \
 	-fsigned-char
 if !CLANG
-	AM_COMMON_CFLAGS += -rdynamic
+	AM_COMMON_CFLAGS += ${RDYNAMIC_FLAG}
 endif
 if SOLARIS
 	AM_COMMON_CFLAGS += -Wno-unused-local-typedefs
 endif
 
-AM_CFLAGS = $(AM_COMMON_CFLAGS) $(HARDENING_CFLAGS)
+AM_CFLAGS = $(AM_COMMON_CFLAGS)
+if LINUX
+AM_CFLAGS += $(HARDENING_CFLAGS)
+endif
 AM_CPPFLAGS = $(AM_COMMON_CPPFLAGS)
 AM_CXXFLAGS = \
 	@AM_CXXFLAGS@ \
 	$(AM_COMMON_CFLAGS) \
 	-ftemplate-depth-1024 \
 	-Wnon-virtual-dtor \
-	-Wno-invalid-offsetof $(HARDENING_CFLAGS)
+	-Wno-invalid-offsetof 
+if LINUX
+AM_CXXFLAGS += $(HARDENING_CFLAGS) 
+endif
 if !CLANG
 	AM_CXXFLAGS += -Wstrict-null-sentinel
 endif
 
-# solaris harding
-if SOLARIS
-	AM_CXXFLAGS += -lssp_nonshared
-endif
```

Comment:
Yes, I removed the hardening tags to linux only. Which was the case for the linker flags already. So it removed the requirement for Solaris to have this extra hardening lib. AIX gcc didn't come by default with this lib so it would require building a custom gcc to support it.